code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -620,20 +496,148 @@
+
+/**
+	public
+
+	Configures the icon of the specified container P.
+
+	required data Data persistent through invocations in the same loop.
+	required R The reagents used to make the item P.
+	required P The container to configure.
+*/
+/datum/chemical_production_mode/proc/ConfigureItem(data, datum/reagents/R, obj/item/reagent_containers/P)
+	if(sprites)
+		P.icon_state = sprites[set_sprite]
+	return
+
+/datum/chemical_production_mode/proc/Synthesize(user, location, datum/reagents/reagents, obj/item/storage/S = null)
+	if(!reagents.total_volume)
+		return
+
+	var/medicine_name = set_name
+	var/count = set_items_amount
+	var/amount_per_item = clamp(reagents.total_volume / count, 0, max_units_per_item)
+	if(!isnull(medicine_name) && length(medicine_name) <= 0)
+		medicine_name = PlaceholderNameWithAmount(reagents, amount_per_item)
+
+	var/data = list()
+	for(var/i = 1 to count)
+		if(reagents.total_volume <= 0)
+			to_chat(user, "<span class='warning'>Not enough reagents to create these items!</span>")
+			return
+
+		var/obj/item/reagent_containers/P = new item_type(location)
+		if(!isnull(medicine_name))
+			P.name = "[medicine_name][name_suffix]"
+		P.pixel_x = rand(-7, 7) // Random position
+		P.pixel_y = rand(-7, 7)
+		reagents.trans_to(P, amount_per_item)
+		ConfigureItem(data, reagents, P)
+
+		// Load the items into the bottle if there's one loaded
+		if(istype(S) && S.can_be_inserted(P, TRUE))
+			P.forceMove(S)
+
+/datum/chemical_production_mode/pills
+	production_name = "Pills"
+	production_icon = "pills"
+	item_type = /obj/item/reagent_containers/pill
+	max_items_amount = 20
+	max_units_per_item = 100
+	name_suffix = " pill"
+
+/datum/chemical_production_mode/pills/New()
+	. = ..()
+	sprites = list()
+	for(var/i = 1 to MAX_PILL_SPRITE)
+		sprites += list("pill[i]")
+
+/datum/chemical_production_mode/patches
+	production_name = "Patches"
+	production_icon = "plus-square"
+	item_type = /obj/item/reagent_containers/patch
+	max_items_amount = 20
+	max_units_per_item = 30
+	name_suffix = " patch"
+
+	var/static/list/safe_chem_list = list("antihol", "charcoal", "epinephrine", "insulin", "teporone", "silver_sulfadiazine", "salbutamol",
+									"omnizine", "stimulants", "synaptizine", "potass_iodide", "oculine", "mannitol", "styptic_powder",
+									"spaceacillin", "salglu_solution", "sal_acid", "cryoxadone", "blood", "synthflesh", "hydrocodone",
+									"mitocholide", "rezadone", "menthol", "diphenhydramine", "ephedrine", "iron", "sanguine_reagent")
+
+/datum/chemical_production_mode/patches/proc/SafetyCheck(datum/reagents/R)
+	for(var/datum/reagent/A in R.reagent_list)
+		if(!safe_chem_list.Find(A.id))
+			return FALSE
+	return TRUE
+
+/datum/chemical_production_mode/patches/ConfigureItem(data, datum/reagents/R, obj/item/reagent_containers/patch/P)
+	var/chemicals_is_safe = data["chemicals_is_safe"]
+
+	if(isnull(chemicals_is_safe))
+		chemicals_is_safe = data["chemicals_is_safe"] = SafetyCheck(R)
```

Comment:
No idea what you're trying to do here. Please use 2 or more separate assignments